### ISSUE TITLE
Error messages not translating

### ISSUE THREAD

**Issue by jyourstone** [woocommerce/woocommerce-gateway-stripe#438]
#### Affected ticket(s)

None

#### What I expected

That when a card is declined for some reason, the error message it displayed in the same language as the Stripe plugin is set to.

#### What happened instead

When a card is declined, the error message appears in English.

#### Steps to reproduce the issue

- Set store language to another than English (Swedish in my case).
- Use a test card that triggers error messages (for example 4000000000000127).

#### Versions I use

WooCommerce Stripe Gateway 4.0.0
Stripe API 2017-12-14
WooCommerce 3.2.6

#### Additional information

These error messages are not part of the Stripe plugin and can't be translated on translate.wordpress.org. I submitted this issue a while back (#194) and it was fixed then, but after upgrading to WooCommerce Stripe Gateway 4.0.0 the error messages appear in English again.
I have confirmed that the error messages are translated in version 3.2.3.

-------------------

- [ ] Issue assigned to next milestone.
- [ ] Issue assigned a priority (will be assessed by maintainers).

**Comment by jyourstone** [comment]
The error messages are still in English, even after updating Stripe to 4.0.1.

**Comment by thorsten-stripe** [comment]
@roykho I think we might be missing the setting of the locale for Stripe Elements[0]: https://stripe.com/docs/stripe-js/reference#stripe-elements
> `locale`: The IETF language tag of the locale to display placeholders and error strings in. Setting the locale does not affect the behavior of postal code validation—a valid postal code for the billing country of the card is still required. Default is 'auto' (Stripe detects the locale of the browser). Supported values are: ar, da, de, en, es, fi, fr, he, it, ja, no, nl, pl, sv, zh.

[0] https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/assets/js/stripe.js#L9

**Comment by roykho** [comment]
@jyourstone this is what I am seeing ![](http://cld.wthms.co/PXsLqR+)

**Comment by jyourstone** [comment]
@roykho this is what I am seeing when using the test card number 4000000000000101 to trigger a cvc_check fail.

![screenshot](https://user-images.githubusercontent.com/4037547/34879086-6b4d6ff4-f7ac-11e7-85c2-1898f0ebe3f3.jpg)

**Comment by roykho** [comment]
@jyourstone please dump your browser cache and restart the browser and try again.

**Comment by roykho** [comment]
@jyourstone Or try downloading the translation files from WP server again. You may need to click update to translation button on your WP update screen.

**Comment by jyourstone** [comment]
@roykho I already did that several times and I'm also using the latest translation files (which doesn't matter because these error messages are not part of the Stripe plugin as discussed previously)

**Comment by roykho** [comment]
@jyourstone yes they are apart of the plugin. They come from Stripe originally but they're mapped within the plugin.

**Comment by jyourstone** [comment]
@roykho Not part of the .po file at least, and if so they are wrong. We had this discussion here #194 
The error message says "YOUR card's..." while the translation file says "THE card's..."

**Comment by roykho** [comment]
@jyourstone since 4.0 we don't use translation files in the plugin anymore. So don't rely on any of the POT files in the plugin. You have to download the latest from the main source which is automatic. So please go to your settings change your language to English click save. Then change it back to Swedish and save. That should trigger a download.

**Comment by jyourstone** [comment]
@roykho Yes I'm talking about the .PO file that downloads from translate.wordpress.org, sorry if I was unclear. There is no string that corresponds to the error message on translate.wordpress.org: https://translate.wordpress.org/projects/wp-plugins/woocommerce-gateway-stripe/stable/sv/default

**Comment by roykho** [comment]
@jyourstone is the translation you see in my screenshot correct in Swedish? I am using the test card you mentioned in your original post 4000000000000127

**Comment by jyourstone** [comment]
@roykho Yes in your screenshot the error message is correct.
It's still not working for me on my staging site though: http://staging.yourstone-keramik.se/kassan/

What could be the issue there?

**Comment by roykho** [comment]
@jyourstone the only thing I can think of at the moment is perhaps the error code/type you're getting is different from mine and thus in my test it works as it is mapped and the one you're seeing hasn't been accounted for. Can you please check your browser console and see if you can spot the error log after you click on place order? You may have to do this in test mode for it to log, I can't remember.

**Comment by jyourstone** [comment]
@roykho I can't find it in the browser console, here's what the WooCommerce log says:

`01-12-2018 @ 16:01:44 - ====Start Log====
Error: Your card's security code is incorrect.
====End `Log====``

**Comment by roykho** [comment]
@jyourstone ok let me research this more and will let you know.

**Comment by jyourstone** [comment]
Thank you, I will also try to find out more information.

**Comment by roykho** [comment]
Just leaving this as a test note that I switched the language to French as a test and I also got the correct results ![](http://cld.wthms.co/sMhomk+)

**Comment by jyourstone** [comment]
I've done some more testing.

- Cleared cache again, tried another browser and a different computer.
- Updated Stripe to 4.0.1 on a completely different WordPress installation.

Error messages still display in English. When I downgrade to 3.2.3 they are translated, but as soon as I update to 4.0.1 they are in English again.

**Comment by roykho** [comment]
Please open up a ticket with us so we can get your site login and possible FTP information so we could debug this on your site https://woocommerce.com/my-account/create-a-ticket

**Comment by jyourstone** [comment]
Submitted with ticket ID #897443

**Comment by roykho** [comment]
@jyourstone thanks for the submitting the ticket. I think I found the issue and why we're seeing different. You have this option off http://cld.wthms.co/CINzpi and I have it on. So because of this, in the code, it takes different paths. I was looking at the path on my site with it on.

This is why I asked for you to submit a ticket because I figured there must be a setting that is different from what I am testing.
